charmcraft.yaml

```diff
@@ -89,3 +89,15 @@ config:
     httpreq_username:
       type: string
       description: Basic authentication username
+    http_proxy:
+      description: URL of the HTTP proxy eg http://proxy.internal:6666, it will set the HTTP_PROXY var in the workload environment
+      type: string
+      default: ''
+    https_proxy:
+      description: URL of the HTTPS proxy eg http://proxy.internal:6666, it will set the HTTPS_PROXY var in the workload environment
+      type: string
+      default: ''
+    no_proxy:
+      description: Domains that need to be excluded from proxying no_proxy="test.com,test.co.uk", it is a comma separate list
+      type: string
+      default: ''
```

Comment:
Just like it is done in GitHub Runners charm and other IS charms, let's use the Juju env variables for this.

Reference
- https://github.com/canonical/github-runner-operator/blob/82067e501bbb9e7e23330c433098b4083beaa433/src/charm_state.py#L674-L697

Reply:
I'm happy to change it

although using those env variables might be counterproductive 

ideally you have 2 layers, one where u use those by default and another where u can override that behaviour

Reply:
(I placed the same comment on the lego k8s lib, let's discuss it over there)

Reply:
I don't have a strong opinion, but I see there was a discussion [here](https://github.com/canonical/httprequest-lego-k8s-operator/issues/125#issuecomment-2177927953). And the point was made to keep consistency across charms.
In addition using the juju model env vars would allow us to handle this in the lib only without the need to change any of the Lego charms.